src/airport.js

```diff
@@ -1 +1,98 @@
+class Airport {
 
+    constructor(airportID) {
+        this.airportID = airportID;
+        this.listOfPlanesAtAirport = [];
+        this.airportMaxCapacity = 2;
+        this.weatherSafeForAction = true;
+    }
+
+    get getAirportID() {
+        return this.airportID;
+    }
+
+    get getListOfPlanesAtAirport() {
+        return this.listOfPlanesAtAirport;
+    }
+
+    get getAirportMaxCapacity() {
+        return this.airportMaxCapacity;
+    }
+
+    setAirportMaxCapacity(newCapacity) {
+        if (newCapacity >= this.numberOfPlanesAtAirport() && typeof newCapacity === 'number' && newCapacity >= 0) {
+            this.airportMaxCapacity = newCapacity
+        }
+    }
+
+
+    instructToLand(plane) {
+        if (this.preLandChecks(plane)) {
+            this.listOfPlanesAtAirport = [...this.listOfPlanesAtAirport, plane];
+            plane.setInFlight(false);
+        }
+    }
+
+    isAirportFull() {
+        return this.airportMaxCapacity === this.getListOfPlanesAtAirport.length;
+    }
+
+    instructToTakeOff2(plane) {
+
+        if (this.preTakeOffChecks(plane)) {
+
+            this.listOfPlanesAtAirport = this.listOfPlanesAtAirport.filter(findPlanes);
+
+            function findPlanes(foundPlane) {
+                if (foundPlane?.getPlaneID() != plane?.getPlaneID()) {
+                    return foundPlane
+                } else {
+                    plane.setInFlight(true)
+                }
+
+            }
+        }
+
+    }
+
+    checkPlaneAtAirport(plane) {
```

Comment:
this could possibly be better. Some() is one or more but the first instance of a duplicate should flagged and returned to help performance and execution time, especially if scaling up the program. Also the program should be written in a way that prevents duplicate planes at an airport as much as possible, not just relying on how things perceived to be coded. I used some() because it returned a TRUE or FALSE, and didn't find the other dot notation covered what I wanted.